src/mlpack/core/tree/binary_space_tree/binary_space_tree_impl.hpp

```diff
@@ -1082,33 +1082,62 @@ void BinarySpaceTree<MetricType, StatisticType, MatType, BoundType, SplitType>::
     right = NULL;
   }
 
-  ar & BOOST_SERIALIZATION_NVP(begin);
-  ar & BOOST_SERIALIZATION_NVP(count);
-  ar & BOOST_SERIALIZATION_NVP(bound);
-  ar & BOOST_SERIALIZATION_NVP(stat);
+  ar & CEREAL_NVP(begin);
+  ar & CEREAL_NVP(count);
+  ar & CEREAL_NVP(bound);
+  ar & CEREAL_NVP(stat);
 
-  ar & BOOST_SERIALIZATION_NVP(parentDistance);
-  ar & BOOST_SERIALIZATION_NVP(furthestDescendantDistance);
-  ar & BOOST_SERIALIZATION_NVP(dataset);
+  ar & CEREAL_NVP(parentDistance);
+  ar & CEREAL_NVP(furthestDescendantDistance);
 
-  // Save children last; otherwise boost::serialization gets confused.
+  // Save children last; otherwise cereal gets confused.
```

Comment:
```suggestion
  // Save children last.
```
Actually I think cereal is not going to get confused, but it's probably still good to serialize the children last.  If I remember right, the reason I had to do that for boost was that its pointer tracking would get very confused.